e2e/__tests__/readInitialOptions.test.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import path = require('path');
+import execa = require('execa');
+import {readInitialOptions} from 'jest-config';
+
+function resolveFixture(...pathSegments: Array<string>) {
+  return path.resolve(__dirname, '..', 'read-initial-options', ...pathSegments);
+}
+
+describe(readInitialOptions, () => {
+  test('should read from the cwd by default', async () => {
+    const configFile = resolveFixture('js-config', 'jest.config.js');
+    const rootDir = resolveFixture('js-config');
+    jest.spyOn(process, 'cwd').mockReturnValue(rootDir);
+    const {config, configPath} = await readInitialOptions();
+    expect(config).toEqual({jestConfig: 'jest.config.js', rootDir});
+    expect(configPath).toEqual(configFile);
+  });
+  test('should read a jest.config.js file', async () => {
+    const configFile = resolveFixture('js-config', 'jest.config.js');
+    const rootDir = resolveFixture('js-config');
+    const {config, configPath} = await readInitialOptions(configFile);
+    expect(config).toEqual({jestConfig: 'jest.config.js', rootDir});
+    expect(configPath).toEqual(configFile);
+  });
+  test('should read a package.json file', async () => {
+    const configFile = resolveFixture('pkg-config', 'package.json');
+    const rootDir = resolveFixture('pkg-config');
+    const {config, configPath} = await readInitialOptions(configFile);
+    expect(config).toEqual({jestConfig: 'package.json', rootDir});
+    expect(configPath).toEqual(configFile);
+  });
+  test('should read a jest.config.ts file', async () => {
+    const configFile = resolveFixture('ts-config', 'jest.config.ts');
+    const rootDir = resolveFixture('ts-config');
+    // Read by proxy, because we're running in a VM and are not allowed to import 'ts-node' directly
+    const {stdout} = await execa('node', ['../readOptions.js'], {
+      cwd: rootDir,
+    });
+    const {config, configPath} = JSON.parse(stdout);
+    expect(config).toEqual({jestConfig: 'jest.config.ts', rootDir});
+    expect(configPath).toEqual(configFile);
+  });
+  test('should read a jest.config.mjs file', async () => {
+    const configFile = resolveFixture('mjs-config', 'jest.config.mjs');
+    const rootDir = resolveFixture('mjs-config');
+    // Read by proxy, because we're running in a VM and are not allowed to `import` directly
+    const {stdout} = await execa('node', ['../readOptions.js'], {
+      cwd: rootDir,
+    });
```

Comment:
For my understanding this approach should be used in all test in this file. It is based on real world implementation and that is how I like e2e tests to be written.

In contrary, importing `readInitialOptions` to a test file and mocking `process.cwd` still feels like a unit test. Of course, that just an approach or opinion. The same thing is tested either way.

Reply:
True, using execa is much better. Fixed with the latest push